packages/truffle-contract/lib/execute.js

```diff
@@ -1,6 +1,7 @@
 const debug = require("debug")("contract:execute"); // eslint-disable-line no-unused-vars
 var Web3PromiEvent = require("web3-core-promievent");
 var EventEmitter = require("events");
+var BigNumber = require("bignumber.js");
```

Comment:
Why bignumber.js instead of BN.js? We really should avoid multiple high-precision number libraries.

I know truffle-contract already has the bignumber.js dependency, but that is only for compatibility reasons, to enable `MyContract.numberFormat` to give BigNumber return values.

Reply:
It was originally `bn.js`, but `constructor.gasMultiple` is/can be a decimal value and `bn.js` does not support decimals anywhere (even as an argument to `.muln()` for example). I could not think of an elegant way to turn an arbitrary decimal into a single multiply/division of two numbers.

Reply:
A hacky way to do it might be:

```javascript
let decimal; // decimal to convert
let maxPrecision; // upperbound on significant figures
let numerator;
let denominator;
const sigfigs = Math.min(
  decimal.toString().length - 1, // length less one because `.`
  maxPrecision
);

denominator = Math.pow(10, sigfigs);
numerator = decimal * denominator;
```

I strongly urge you to hack your way away from using bignumber.js for this case. Don't let decimals stop you when there are large powers of ten to be wielded.

Reply:
I definitely find this hack gross, but I will implement it.